File: frontend/src/api/kbRequests.ts
import { api } from './client'

export interface KbRequestRecord {
  id: number
  requester_user_id: string
  requester_username: string
  requester_organization: string
  requested_name: string
  requested_description: string
  request_reason: string
  status: 'pending' | 'approved' | 'rejected' | 'created' | 'failed'
  reviewer_user_id: string
  reviewer_username: string
  review_reason: string
  approved_kb_id: string
  approved_kb_name: string
  create_error: string
  created_at: string
  updated_at: string
}

export interface SubmitKbRequestBody {
  requested_name: string
  requested_description: string
  request_reason: string
}

export const kbRequestApi = {
  async submit(body: SubmitKbRequestBody): Promise<KbRequestRecord> {
    const resp = await api.post('/kb-requests', body)
    return resp.data
  },
  async mine(page = 1, pageSize = 20): Promise<{ items: KbRequestRecord[]; total: number; page: number; page_size: number }> {
    const resp = await api.get('/kb-requests/mine', { params: { page, page_size: pageSize } })
    return resp.data
  },
  async list(page = 1, pageSize = 20, status = ''): Promise<{ items: KbRequestRecord[]; total: number; page: number; page_size: number }> {
    const resp = await api.get('/admin/kb-requests', { params: { page, page_size: pageSize, status } })
    return resp.data
  },
  async approve(id: number): Promise<KbRequestRecord> {
    const resp = await api.post(`/admin/kb-requests/${id}/approve`)
    return resp.data
  },
  async reject(id: number, reason = ''): Promise<KbRequestRecord> {
    const resp = await api.post(`/admin/kb-requests/${id}/reject`, { reason })
    return resp.data
  },
}

File: backend/app/api/v1/kb_requests.py
"""知识库申请路由。"""
from pydantic import BaseModel, Field
from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession

from app.api.v1.auth import UserInfo, get_current_user
from app.core.db import get_session
from app.services.kb_request_service import KbRequestError, create_request, list_my_requests

router = APIRouter(prefix="/api/kb-requests", tags=["kb-requests"])


class KbRequestCreateBody(BaseModel):
    """创建知识库申请请求体。"""

    requested_name: str = Field(min_length=1, max_length=255)
    requested_description: str = ""
    request_reason: str = ""
    want_wiki: bool = False
    want_llm_graph: bool = False
    extract_focus: str = ""
    relation_types: str = ""
    example_text: str = ""


@router.post("")
async def submit_kb_request(
    body: KbRequestCreateBody,
    user: UserInfo = Depends(get_current_user),
    session: AsyncSession = Depends(get_session),
) -> dict:
    """提交知识库申请。"""
    try:
        record = await create_request(
            session,
            user,
            body.requested_name,
            body.requested_description,
            body.request_reason,
            body.want_wiki,
            body.want_llm_graph,
            body.extract_focus,
            body.relation_types,
            body.example_text,
        )
    except KbRequestError as exc:
        from fastapi import HTTPException

        raise HTTPException(status_code=exc.status_code, detail=exc.message)
    return {
        "id": record.id,
        "requester_user_id": record.requester_user_id,
        "requester_username": record.requester_username,
        "requester_organization": record.requester_organization,
        "requested_name": record.requested_name,
        "requested_description": record.requested_description,
        "request_reason": record.request_reason,
        "want_wiki": bool(record.want_wiki),
        "want_llm_graph": bool(record.want_llm_graph),
        "extract_focus": record.extract_focus,
        "relation_types": record.relation_types,
        "example_text": record.example_text,
        "status": record.status,
        "reviewer_user_id": record.reviewer_user_id,
        "reviewer_username": record.reviewer_username,
        "review_reason": record.review_reason,
        "approved_kb_id": record.approved_kb_id,
        "approved_kb_name": record.approved_kb_name,
        "create_error": record.create_error,
        "created_at": record.created_at,
        "updated_at": record.updated_at,
    }


@router.get("/mine")
async def my_kb_requests(
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=100),
    user: UserInfo = Depends(get_current_user),
    session: AsyncSession = Depends(get_session),
) -> dict:
    """获取我的知识库申请列表。"""
    return await list_my_requests(session, user, page, page_size)
